src/providers/activeLearning/objectDetection.ts

```diff
@@ -0,0 +1,254 @@
+                this.jsonClasses = JSON.parse(JSON.stringify(response.data));
+            } else {
+                const handler = new ElectronProxyHandler(modelFolderPath);
+                this.model = await tf.loadGraphModel(handler);
+                this.jsonClasses = await handler.loadClasses();
+            }
+
+            // Warmup the model.
+            const result = await this.model.executeAsync(tf.zeros([1, 300, 300, 3])) as tf.Tensor[];
+            result.forEach(async (t) => await t.data());
+            result.forEach(async (t) => t.dispose());
+            this.modelLoaded = true;
+        } catch (err) {
+            this.modelLoaded = false;
+            throw err;
+        }
+    }
+
+    /**
+     * Predict Regions from an HTMLImageElement returning list of IRegion.
+     * @param image ImageObject to be used for prediction
+     * @param predictTag Flag indicates if predict only region bounding box of tag too.
+     * @param xRatio Width compression ratio between the HTMLImageElement and the original image.
+     * @param yRatio Height compression ratio between the HTMLImageElement and the original image.
+     */
+    public async predictImage(image: ImageObject, predictTag: boolean, xRatio: number, yRatio: number)
+        : Promise<IRegion[]> {
+        const regions: IRegion[] = [];
+
+        const predictions = await this.detect(image);
+        predictions.forEach((prediction) => {
+            const left = Math.max(0, prediction.bbox[0] * xRatio);
+            const top = Math.max(0, prediction.bbox[1] * yRatio);
+            const width = Math.max(0, prediction.bbox[2] * xRatio);
+            const height = Math.max(0, prediction.bbox[3] * yRatio);
+
+            regions.push({
+                id: shortid.generate(),
+                type: RegionType.Rectangle,
+                tags: predictTag ? [prediction.class] : [],
+                boundingBox: {
+                    left,
+                    top,
+                    width,
+                    height,
+                },
+                points: [{
+                    x: left,
+                    y: top,
+                },
+                {
+                    x: left + width,
+                    y: top,
+                },
+                {
+                    x: left + width,
+                    y: top + height,
+                },
+                {
+                    x: left,
+                    y: top + height,
+                }],
+            });
+        });
+
+        return regions;
+    }
+
+    /**
+     * Detect objects for an image returning a list of bounding boxes with
+     * associated class and score.
+     *
+     * @param img The image to detect objects from. Can be a tensor or a DOM
+     *     element image, video, or canvas.
+     * @param maxNumBoxes The maximum number of bounding boxes of detected
+     * objects. There can be multiple objects of the same class, but at different
+     * locations. Defaults to 20.
```

Comment:
We don't have to do the work now, but 20 sounds like it could be handy as a setting...

Reply:
Agreed - will take this as an update with some other settings like thresholds.